### PR TITLE
ParseEmailFiles fails on demisto.debug line

### PR DESCRIPTION
**Pull request by tallieber**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [41085](https://github.com/demisto/etc/issues/41085)

## Description
We use demisto.debug to print an error where a string could not be decoded. the problematic strings contains characters that raised a server error: `Error from Scripts is : Script failed to run: 
Error: Failed to decode (loop) data from docker code script: invalid character '{' after object key:value pair
This may happen when you don't return an explicit entry (or entries) from a script.`
Will open an issue for this problem.

### CODE REVIEW

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py [hunk 1]
```diff
@@ -3432,8 +3432,7 @@ def convert_to_unicode(s, is_msg_header=True):
                 try:
                     res += decoded_s.decode(encoding).encode('utf-8')
                 except UnicodeDecodeError:
-                    demisto.debug('Failed to decode encoded_string:'
-                                  'encoding: {}, encoded_text: {}'.format(encoding, decoded_s))
+                    demisto.debug('Failed to decode encoded_string with encoding {}'.format(encoding))
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py:3435]: lets log here `decoded_s.decode(encoding, errors='replace').encode('utf-8')`

**Review by Itay4 — COMMENTED** [review]
please add test to the test playbook that reproduces the issue

**Comment by tallieber** [comment]
@Itay4 I added a test playbook, in the playbook we see that there is a header that stays encoded (`TestSegurança ?� =?UTF-8?Q?Informação`). 
The value we had prior  to 1_4_26 release is `Test =?UTF-8?Q?Seguran=C3=A7a=20?=da =?UTF-8?Q?Informa=C3=A7=C3=A3o?=`, not decode at all.
We revert most of the code on 1_4_28 ([PR](https://github.com/demisto/content/pull/14585)) and now will handle some cases with replace as in the first line.  
So for now as we said in the revert we will not handle this kind of encoded text and we should handle it properly when re-write this code. 
For the error which raised by this bug: [41085](https://github.com/demisto/etc/issues/41085), this PR is taking care of it, and the test playbook is relevant for it too.

File: Packs/CommonScripts/ReleaseNotes/1_4_33.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Scripts
+##### ParseEmailFiles
+- Fixed an issue where ParseEmailFiles fails while print error to log.
```
Comment (Itay4) [inline Packs/CommonScripts/ReleaseNotes/1_4_33.md:4]:
```suggestion
- Fixed an issue where the automation script failed to process MIME encoded-words.
```

File: TestData/test-subject-raised-error-on-log.eml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Subject: Test
```
Comment (Itay4) [inline TestData/test-subject-raised-error-on-log.eml:1]: this should be in the `test_data` subdir of the script dir

File: Packs/CommonScripts/TestPlaybooks/playbook-ParseEmailFiles-test.yml [hunk 4]
```diff
@@ -804,12 +783,129 @@ tasks:
...
+    skipunavailable: false
+    quietmode: 0
+  "25":
+    id: "25"
+    taskid: 8e82f75f-87d0-48fc-8eed-6b829b328a57
+    type: regular
+    task:
+      id: 8e82f75f-87d0-48fc-8eed-6b829b328a57
+      version: -1
+      name: Subject with special chars
```
Comment (Itay4) [inline Packs/CommonScripts/TestPlaybooks/playbook-ParseEmailFiles-test.yml:821]:
```suggestion
      name: Organization header with MIME encoded-word
```

**Review by Itay4 — APPROVED** [review]
please note my comments

**Pull request merged**